Integrations/MicrosoftGraphListener/MicrosoftGraphListener.py

```diff
@@ -0,0 +1,986 @@
+                           resp_type="text")
+
+        message_content.pop('attachments', None)
+        message_content.pop('internet_message_headers', None)
+        human_readable = tableToMarkdown(f'Email was sent successfully.', message_content)
+        ec = {self.CONTEXT_SENT_EMAIL_PATH: message_content}
+
+        return human_readable, ec
+
+    def reply_to(self, to_recipients, comment, message_id):
+        """
+        Sends reply message to recipients.
+
+        :type to_recipients: ``list``
+        :param to_recipients: List of recipients to reply.
+
+        :type comment: ``str``
+        :param comment: The comment to send as a reply
+
+        :type message_id: ``str``
+        :param message_id: The message id to reply.
+
+        :return: String representation of markdown message regarding successful message submission.
+        rtype: ``str``
+        """
+        suffix_endpoint = f'/users/{self._mailbox_to_fetch}/messages/{message_id}/reply'
+        reply = MsGraphClient._build_reply(to_recipients, comment)
+        self._http_request('POST', suffix_endpoint, json_data=reply, resp_type="text")
+
+        return f'### Replied to: {", ".join(to_recipients)} with comment: {comment}'
+
+    def send_draft(self, draft_id):
+        """
+        Send draft message.
+
+        :type draft_id: ``str``
+        :param draft_id: Draft id to send.
+
+        :return: String representation of markdown message regarding successful message submission.
+        :rtype: ``str``
+        """
+        suffix_endpoint = f'/users/{self._mailbox_to_fetch}/messages/{draft_id}/send'
+        self._http_request('POST', suffix_endpoint, resp_type="text")
+
+        return f'### Draft with: {draft_id} id was sent successfully.'
+
+    def test_connection(self):
+        """
+        Basic connection test instead of test-module.
+
+        :return: Returns markdown string representation of success or Exception in case of login failure.
+        rtype: ``str`` or Exception
+        """
+        suffix_endpoint = f'users/{self._mailbox_to_fetch}'
+        user_response = self._http_request('GET', suffix_endpoint)
+
+        if user_response.get('mail') != '' and user_response.get('id') != '':
+            return_outputs('```✅ Success!```')
+        else:
+            raise Exception("Failed validating the user.")
+
+
+def main():
+    """ COMMANDS MANAGER / SWITCH PANEL """
+    params = demisto.params()
+    # params related to oproxy
+    auth_id = params.get('auth_id')
+    # In case the script is running for the first time, refresh token is retrieved from integration parameters,
+    # in other case it's retrieved from integration context.
+    refresh_token = demisto.getIntegrationContext().get('current_refresh_token') or params.get('refresh_token', '')
+    enc_key = params.get('enc_key')
+    token_retrieval_url = 'https://oproxy.demisto.ninja/obtain-token'  # disable-secrets-detection
```

Comment:
Token retrieval should be configurable via the auth_id param (using `@`) so it is possible to use dev instance.. See: https://github.com/demisto/content/blob/master/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py#L26